data/serializer/src/main/java/tech/pegasys/teku/api/migrated/TotalAttestationReward.java

```diff
@@ -34,6 +38,29 @@ public TotalAttestationReward(
     this.inclusionDelay = inclusionDelay;
   }
 
+  public TotalAttestationReward(long validatorIndex, final RewardAndPenalty rewardAndPenalty) {
+    this.validatorIndex = validatorIndex;
+
+    final DetailedRewardAndPenalty detailedRewardAndPenalty =
+        rewardAndPenalty
+            .asDetailed()
+            .orElseThrow(
+                () ->
+                    new InvalidParameterException(
+                        "TotalAttestationRewards requires a DetailedRewardAndPenalty instance"));
+
+    this.head =
+        detailedRewardAndPenalty.getReward(RewardComponent.HEAD).longValue()
+            - detailedRewardAndPenalty.getPenalty(RewardComponent.HEAD).longValue();
+    this.target =
+        detailedRewardAndPenalty.getReward(RewardComponent.TARGET).longValue()
+            - detailedRewardAndPenalty.getPenalty(RewardComponent.TARGET).longValue();
+    this.source =
+        detailedRewardAndPenalty.getReward(RewardComponent.SOURCE).longValue()
+            - detailedRewardAndPenalty.getPenalty(RewardComponent.SOURCE).longValue();
+    this.inclusionDelay = Optional.empty();
+  }
```

Comment:
maybe our `DetailedRewardAndPenalty` should just store long's? they could be set as uint, and just store the long value and we'd not need to unbox for responses etc...